### DISCUSSION TITLE
Plausible 0 visitors wordpress

### DISCUSSION THREAD

**Discussion by LordPaulHD** [plausible/analytics#3819] [Self-Hosted Support]
Hi, I just finished to set up a self hosted plausible server. I use nginx as an reverse proxy.
But when I use the official wordpress plugin to connect to my self hosted instance (plausible.myserver.de) and leave the rest of the configuration to standard, I don't get a visitors output.

I found somebody with a similar problem just he had a code snippet for a normal website. His solution was to add data-api="https://plausible.myserver.de/api/event" to the code snippet. I tried this and deactivated the plugin and it works.

However i would prefer to use the actual wordpress plugin. Is there a way I can make it work?

Thanks for answers
Paul

**Comment by rossdav70** [comment]
Plugin works fine for my self hosted sites. Silly question, but have you set the address in the self hosted tab? And of course have you checked incognito as by default logged in users aren't counted

**Comment by ruslandoga** [comment]
👋 @LordPaulHD 

Please share more information on your setup :) What versions do you use, configs, do you use any proxies, etc.

**Comment by LordPaulHD** [comment]
Hello, i wanted to say upfront that I'm not an native english speaker and very tired so there are going to be a lot of grammar and spelling mistakes in this and I'm very sorry about it. I hope everything is understandable. 

@rossdav70 
Yes i have filled it out at the self hosted site multiple times and checked from not logged in browser on my PC, Iphone and Ipad. Do you use a reversed proxy for your plausible server?

@ruslandoga 
I'm sorry I have in fact not given a lot of information in my original question.
I use a V-server running ubuntu 20.04
On the V-servers i of course run Docker. 
Here my docker-compose.yml:
`version: "3.3"
services:

  nginx:
    image: nginx:alpine
    container_name: nginx
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - ./certbot/conf:/etc/letsencrypt 
      - ./certbot/www:/var/www/certbot
    networks:
      - plausible

  certbot:
    image: certbot/certbot
    container_name: certbot
    volumes:
      - ./certbot/conf:/etc/letsencrypt
      - ./certbot/www:/var/www/certbot
    command: certonly --webroot -w /var/www/certbot --force-renewal --email me@my-email.de -d plausible.myplauserver.de --agree-tos

  mail:
    image: bytemark/smtp
    restart: always
    networks:
      - plausible

  plausible_db:
    # supported versions are 12, 13, and 14
    image: postgres:14-alpine
    restart: always
    volumes:
      - db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_PASSWORD=postgres
    networks:
      - plausible

  plausible_events_db:
    image: clickhouse/clickhouse-server:23.3.7.5-alpine
    restart: always
    volumes:
      - event-data:/var/lib/clickhouse
      - ./clickhouse/clickhouse-config.xml:/etc/clickhouse-server/config.d/logging.xml:ro
      - ./clickhouse/clickhouse-user-config.xml:/etc/clickhouse-server/users.d/logging.xml:ro
    ulimits:
      nofile:
        soft: 262144
        hard: 262144
    networks:
      - plausible

  plausible:
    image: plausible/analytics:v2.0
    container_name: plausible
    restart: always
    command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
    depends_on:
      - plausible_db
      - plausible_events_db
      - mail
    ports:
      - 127.0.0.1:8000:8000
    env_file:
      - plausible-conf.env
    networks:
      - plausible

volumes:
  db-data:
    driver: local
  event-data:
    driver: local

networks:
  plausible:
    driver: bridge`

As you can see i have Nginx, plausible and certbot running as Dockers.
My Plausible config is 
`BASE_URL=https://plausible.myplauserver.de
SECRET_KEY_BASE= my secret key
DISABLE_REGISTRATION=invite_only`

I used Nginx as a reverse proxy and certbot as ssl certificate with the following config:
`server {
  listen 80;
  server_name plausible.myplauserver.de;
  resolver 127.0.0.11;
  return 301 https://$host$request_uri;
}

server {
  listen 443 ssl http2;
  ssl_certificate     /etc/letsencrypt/live/plausible.zylon-digital.de/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/plausible.zylon-digital.de/privkey.pem;
  server_name plausible.myplauserver.de

  location ~ /.well-known/acme-challenge/ {
    root /var/www/certbot;
  }

  location  / {
    proxy_pass http://plausible:8000;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}`

This is basically my whole configuration. all server none public stuff like domains i just made some up. But they are in the format as above, so whenever there is a https:// before there is one in the config and if there isn't there isn't in the config.

To describe the problem a bit further. Whenever i just write plausible.myplauserver.de in the wordpress plugin under self hosted it doesn't report it to my plausible interface. I then tried the normal code snippet i got from plausible which was

`<script defer data-domain="mywebsite.de" src="https://plausible.myplauserver.de/js/script.js"></script>`

That did not work either. However i found something on with a similar problem, however they didn't have a wordpress site but a normal html site and they resolved it by rewriting the code snippet like this:

`<script defer data-api="https://plausible.myplauserver.de /api/event" data-domain="mywebsite.de" src="https://plausible.myplauserver.de/js/script.js"></script>`

whenever I did this it worked with the code snippet. However I would rather get it to work with the actual wordpress plugin due though multiple reasons.

A solution would be greatly appreciated. Thank you guys so much for helping. I got to be honest other that that plausible is amazing.

**Comment by rossdav70** [comment]
@LordPaulHD  ... yes, I am running a Reverse Proxy. The thing is, when my config wasn't right I was only getting a 502 Bad Gateway. My nginx config is different to yours, but the fact you have the front end working suggests to me that it's ok,

When I look at one of my sites that uses the plugin I see this.

<script defer="" data-domain="mysite.com" data-api="https://analytics.mydomain.com/api/event" src="https://analytics.mydomain.com/js/plausible.outbound-links.tagged-events.js?ver=2.0.5" id="plausible"></script>

As you can see the data-api is already added for me. 

Here are my screenshots from within the plugin.


![settings](https://github.com/plausible/analytics/assets/49723362/ca1e2c2c-f5b0-437f-9cad-522aff4ba356)
![self-hosted](https://github.com/plausible/analytics/assets/49723362/d378127b-76ce-4128-b0fc-7ae13e2eacfe)

**Reply by LordPaulHD** [reply]
Hey @rossdav70 ,
Where exactly are you seeing this script? Ahhh, i didn’t think about checking the script. I will do that later and see what mine says.

**Reply by rossdav70** [reply]
In the head section of the page.  Thats what the plugin does, it just injects the correct script based on the options chosen in the settings screen. You really dont need it. Its just for convenience

**Comment by LordPaulHD** [comment]
@rossdav70 
ok I just checked and now it does get interesting. I just redid all three of my websites the exact same way and it works for one. However I checked the source code for the other two and it says the following.

`<script defer data-domain='' data-api='https://plausible.myplauserver.de//api/event' src="[https://plausible.myplauserver.de//js/plausible.outbound-links.js?ver=2.0.5](view-source:https://plausible.myplauserver.de//js/plausible.outbound-links.js?ver=2.0.5)" id="plausible"></script>`

This makes the problem very obvious. data-domain is missing. However I have no glue how to fix that:)

Any Ideas from your side?  You are right it is just for convenience but I do like my convenience and I'm going to have to do this a lot more often and want it to be easily adjustable.

**Reply by rossdav70** [reply]
I've really no idea. I've just taken a  look at the source for the plugin and in filters.php I see the script being output.

```php
if ( 'plausible-analytics' !== $handle ) {
			return $tag;
		}

		$settings    = Helpers::get_settings();
		$api_url     = Helpers::get_data_api_url();
		$domain_name = esc_html( $settings[ 'domain_name' ] );

		// We need the correct id attribute for IE compatibility.
		$tag    = preg_replace( "/\sid=(['\"])plausible-analytics-js(['\"])/", " id=$1plausible$2", $tag );
		$params = "defer data-domain='{$domain_name}' data-api='{$api_url}'";

``` 
Here you can see the $domain_name is being populated from the Domain Name field in the settings panel and then written to the $param variable which is used later when it writes the script out. 

Therefore if your data-domain is blank it must be that your settings aren't including the domain name. Maybe compare the settings on the one that works to the ones that dont. 

Obviously I may be well off here and the section of code I found may have nothing to do with it.
